### PR TITLE
Use Nucleus api + Support multiple commands

### PR DESCRIPTION
**Pull request by clienthax**
Maintains support for legacy command string as well as supporting this
```
ten-hours {
    activetime=36000
    commands=[
    	"broadcast &b<player>&f has played on our server for &b10hours&f!",
    	"lp user <player> parent add regular"
	]
}
```

Removes need for jar in libs and uses api instead

### CODE REVIEW

**Comment by ziceptor** [comment]
@clienthax perhaps it is necessary to add changes in
https://github.com/SimonFlash/ActiveTime/blob/master/src/main/resources/assets/activetime/milestones.conf

File: src/main/java/com/mcsimonflash/sponge/activetime/objects/Milestone.java [hunk 1]
```diff
@@ -6,23 +6,28 @@
 import org.spongepowered.api.Sponge;
 import org.spongepowered.api.entity.living.player.Player;
 
+import java.util.ArrayList;
```
Comment (SimonFlash) [inline src/main/java/com/mcsimonflash/sponge/activetime/objects/Milestone.java]: Doesn't look like this import is used.

File: src/main/java/com/mcsimonflash/sponge/activetime/objects/Milestone.java [hunk 2]
```diff
@@ -6,23 +6,28 @@
...
     }
 
     public void process(Player player, int time) {
         if (time >= activetime && !Storage.hasMilestone(player.getUniqueId(), name)) {
             if (Storage.setMilestone(player.getUniqueId(), name, true)) {
-                String modifiedCommand = command.replace("<player>", player.getName()).replace("<activetime>", Integer.toString(time));
-                Util.createTask("ActiveTime GiveMilestone Sync Processor (" + player.getName() + ")", task -> Sponge.getCommandManager().process(Sponge.getServer().getConsole(), modifiedCommand), 0, false);
+                for (String command : commands) {
+                    String modifiedCommand = command.replace("<player>", player.getName()).replace("<activetime>", Integer.toString(time));
+                    Util.createTask("ActiveTime GiveMilestone Sync Processor (" + player.getName() + ")", task -> Sponge.getCommandManager().process(Sponge.getServer().getConsole(), modifiedCommand), 0, false);
```
Comment (SimonFlash) [inline src/main/java/com/mcsimonflash/sponge/activetime/objects/Milestone.java]: We should be using a single task to process all of the commands here. `Integer.toString(time)` should be updated to `Util.printTime(time)` while we're here.

**Comment by clienthax** [comment]
Havn't had a chance to look at this just yet, will sort soon :)

**Comment by SimonFlash** [comment]
After a bit of debate, what I've decided to do is close this PR so I'm not overwriting things and implement multiple commands in the Milestone revision for ActiveTime 1.4. I probably should have done this at the start to save both of us time, so apologies for having it drag on. I have not found a method to check if the AFK module is loaded using the Nucleus API alone, but I'll keep this in mind and look into it in the future.

For future reference to anyone, it's best to talk to me about potential PRs before getting involved to see if it's something that needs a PR or could simply be handled as an issue instead.

**Comment by clienthax** [comment]
@SimonFlash https://github.com/NucleusPowered/Nucleus/blob/sponge-api/7/nucleus-api/src/main/java/io/github/nucleuspowered/nucleus/api/NucleusAPI.java#L87

Thats the safe way to do it, allows checking if its there too

**Comment by SimonFlash** [comment]
I've already been using it. The issue is that I need a method to check if the AFK module is loaded. Based on past experience, it's possible for the AFK module to be enabled yet the `NucleusAfkService` is unavailable - I'm not sure how or why, but I do need to differentiate the two.

**Comment by ziceptor** [comment]
You can discuss this with @dualspiral

**Comment by dualspiral** [comment]
> I have not found a method to check if the AFK module is loaded using the Nucleus API alone, but I'll keep this in mind and look into it in the future.

The `NucleusAFKService` should always be available once the afk module has loaded. If you request it and it returns an empty optional, its not loaded. Nucleus 1.3 will streamline the loading so should catch potential problems. 

Remember that you can only obtain it after Nucleus finishes loading modules which is during early post-init. So, make sure you do your check beyond post init.

> I have not found a method to check if the AFK module is loaded using the Nucleus API alone

https://github.com/NucleusPowered/Nucleus/blob/sponge-api/7/nucleus-api/src/main/java/io/github/nucleuspowered/nucleus/api/service/NucleusModuleService.java#L22

**Comment by SimonFlash** [comment]
I remember this being an issue a couple months ago where the AFK service could not be found even if the AFK module was loaded (see [ActiveTime.java](https://github.com/SimonFlash/ActiveTime/commit/bbf8a59a44ed4f8d676357dfd6cac4db31b7856f#diff-492925f07e40104c79716c8cd028badeR119) and [NucleusIntegration.java](https://github.com/SimonFlash/ActiveTime/commit/bbf8a59a44ed4f8d676357dfd6cac4db31b7856f#diff-2a928fbfbb0a801317b74a8a24324a29R44) for the commit changes). This was in the `PostInitialization` phase as well, so I believe the AFK service would exist by then if the module was enabled. I'll be switching over to the NucleusAPI in the meantime and keep an eye out for any similar issues so I can gather up some information.

`NucleusModuleService` I probably should have found - I was looking for something like `NucleusAPI.isModuleLoaded("afk")` that delegated to `Nucleus#isModuleLoaded`. Thanks!

**Comment by dualspiral** [comment]
Yep, it’s possible it was missed, or some odd error. Turns out the mute service was never registered and no-one realised...

**Pull request closed without merging**